src/server/routes/entity/entity.tsx

```diff
@@ -84,67 +84,68 @@ export function displayEntity(req: PassportRequest, res: $Response) {
 
 	let editorEntityVisitPromise;
 	if (resLocals.user) {
-		editorEntityVisitPromise = new EditorEntityVisits({
-			bbid: resLocals.entity.bbid,
-			editorId: resLocals.user.id
-		})
-			.save(null, {method: 'insert'})
-			.then(() => achievement.processPageVisit(orm, resLocals.user.id))
-			.catch(
-				// error caused by duplicates we do not want in database
-				() => new Promise(resolve => resolve(false))
-			);
+		try {
+			await new EditorEntityVisits({
+				bbid: resLocals.entity.bbid,
+				editorId: resLocals.user.id
+			})
+				.save(null, {method: 'insert'});
+
+			editorEntityVisitPromise = await achievement.processPageVisit(orm, resLocals.user.id);
+		}
+		catch {
+			// error caused by duplicates we do not want in database
+			return false;
```

Comment:
I don't think we want to `return` here; the original code ended up setting editorEntityVisitPromise to a value, so in this case it should just be:
```suggestion
			editorEntityVisitPromise = false;
```